### ISSUE TITLE
Is porting this to the new kfd exploit possible?

### ISSUE THREAD

**Issue by KernelFox0** [zhuowei/WDBRemoveThreeAppLimit#20]
KFD: https://github.com/felix-pb/kfd

Apparently it can emulate MDC and it also can overwrite some files (as seen in [Posi0nKFD](https://github.com/GenericCoding/kfd))

That would support versions above 16.1.2 (up to 16.5 and 16.6 beta 1)

**Comment by GenericCoding** [comment]
we need a new method to patch installd afaik which kfd currently doesn't have publically.

**Comment by KernelFox0** [comment]
> we need a new method to patch installd afaik which kfd currently doesn't have publically.

What about emulating MDC?

**Comment by zhuowei** [comment]
I tried using Kfund's KFD->MacDirtyCow emulation ([funVnodeOverwriteFile](https://github.com/wh1te4ever/kfund/blob/972651c0b4c81098b844b29d17741cb445772c74/kfd/fun/vnode.h#L67)) yesterday. The resulting executable crashed the first time it accessed a modified rebase page, exactly like what the comment in Kfund said. So it looks like emulating MDC with KFD might not work for this.

Anyways, this is no longer needed after Trollstore 2 was released, as every iOS version that supports KFD also supports Trollstore

**Comment by doraorak** [comment]
> I tried using Kfund's KFD->MacDirtyCow emulation ([funVnodeOverwriteFile](https://github.com/wh1te4ever/kfund/blob/972651c0b4c81098b844b29d17741cb445772c74/kfd/fun/vnode.h#L67)) yesterday. The resulting executable crashed the first time it accessed a modified rebase page, exactly like what the comment in Kfund said. So it looks like emulating MDC with KFD might not work for this.
> 
> Anyways, this is no longer needed after Trollstore 2 was released, as every iOS version that supports KFD also supports Trollstore

This is needed in order to install swift playground on iphones. Sadly my version does not support mdc and seeing that kfd port did not work is kind of sad. Would the so called-mdc2 exploit work for this in theory ? (Idk what it is really just assumed they were similiar because the names are the same)

**Comment by zhuowei** [comment]
> This is needed in order to install swift playground on iphones
Have you tried installing Swift Playgrounds with Trollstore instead?

**Comment by doraorak** [comment]
Yes it crashes on launch. Appearently this bypasses info.plist signature checks so thats why its necessary. Here is a link https://www.reddit.com/r/jailbreak/comments/168to04/run_swift_playgrounds_on_iphone_with_mdc/

**Comment by zhuowei** [comment]
@doraorak I think at this point you're better off waiting for a jailbreak, which would let you use AppSync Unified or similar to bypass the installd checks instead.
